enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -199,214 +153,90 @@ rules:
 
 ```
 
-So, how does an end user discover and utilize our new CR?
-
-First, a few considerations around discovery:
-
-- one option for cluster admins is to allow "low level" users to get/list `Shares`; i.e. `oc get shares`;
- those users can then discover and use them without further involvement from the cluster admin; i.e. the RBAC example above
- is applied to those "low level" users
-- if this is not acceptable, then some sort of exchange needs to occur between the user and cluster admin
-- that exchange results in either cluster admin providing "just enough" information to the user to update their objects so
-the underlying, associated Pod references CSI volumes using `Shares`
-- or the cluster admin modifies the user's object(s) so they get their `Shares`
-
-Second, a few considerations around how we update objects, so they "get" `Shares` associated with them:
-
-- Pod objects themselves, or any API object that exposes the Pod's `volumes` array as part of providing an API that is
-converted by a controller into a Pod, can directly add a volume of the CSI type with the correct metadata in the volumeAttributes, a la
-
-```yaml
-csi:
-  driver: projected-resources.storage.openshift.io
-    volumeAttributes:
-      share: the-projected-resource
-
-```
-
-- with it as part of a Pod for example:
-
- ```yaml
-apiVersion: v1
-kind: Pod
-metadata:
-  name: some-pod
-spec:
-  containers:
-    ...
-     volumeMounts:
-        - name: shared-item-name
-          mountPath: /path/to/shared-item-data
-  volumes:
-      - name: entitlement
-        csi:
-          driver: projectedresources.storage.openshift.io
-          volumeAttributes:
-             share: the-projected-resource
-
-```
-
-- API objects which do not expose the resulting Pod's volume array will need to have an annotation set on it which triggers
-an admission plugin to mutate the Pod and add the CSI volume.
-
-On that new admission plugin introduced and its validation/mutation of Pods:
-- Pods with this annotation are mutated, adding the CSI driver volume specification,
-- if the Pod `ServiceAccount` is allowed sufficient
-access to the requested `ProjectedResource` (i.e. a SAR check for the SA getting the requested `ProjectedResource` passes)
-- if a Pod already happens to have the CSI Volume specified with reference to a `ProjectedResource`, again the plugin will see if the
-associated `ServiceAccount` for the Pod has sufficient access to the `ProjectedResource`
-- if sufficient access does not exist, the admission plugin marks the Pod as invalid/forbidden.
-- if sufficient access does exists, the admission plugin adds a CSI volume into the Pod's `volume` array, to match the
-example noted above.
-
+The cluster admin is then responsible for assigning this cluster role to appropriate serivce accounts that consume the share (either directly or via role aggregation).
```

Comment:
Shortened the RBAC requirements for brevity. We assume a cluster admin knows how to do this.

Reply:
LGTM